pandas/core/dtypes/cast.py

```diff
@@ -1747,3 +1747,32 @@ def validate_numeric_casting(dtype: np.dtype, value):
     ):
         if is_bool(value):
             raise ValueError("Cannot assign bool to float/integer series")
+
+
+def safe_convert_to_ndarray(values) -> np.ndarray:
+    """
+    Convert values to ndarray while casting nullable dtype arrays to float.
+
+    Parameters
+    ----------
+    values
+        Series or array.
+
+    Returns
+    -------
+    converted_values : np.ndarray
+        Values cast to np.ndarray.
+    """
+    if hasattr(values, "dtype") and is_extension_array_dtype(values.dtype):
+        if is_integer_dtype(values.dtype):
+            converted_values = values.to_numpy(dtype=float, na_value=np.nan)
+        elif is_bool_dtype(values.dtype):
+            converted_values = values.to_numpy(dtype=float, na_value=np.nan)
+        elif is_datetime64tz_dtype(values.dtype):
+            print(values)
+            converted_values = np.asarray(values.dt.tz_localize(tz=None))
+        else:
```

Comment:
Do any of your test cases require `.dt.` here? You write

> the problem here is that for a Series object it'd need to call Series.dt.tz_localize

but it looks to me like `values` won't be a `Series` for any of the test cases you've added - is there an existing test case for which it would?

Reply:
Yes, in the `test_nullable_int_plot` that I added. It tests that it fixes the original bug (plotting with nullable integers) and here the common usage would end up passing a `Series`. I now changed the `_compute_plot_data` method to only pass the `values` and so we can drop support for `Series` in this function.

Reply:
Sorry, what I meant was - is there any test that would hit this branch (`elif is_datetime64tz_dtype(values.dtype):`)?

Reply:
No, in https://github.com/pandas-dev/pandas/blob/8866c961ef15fd0358e11dd429ab41456168849c/pandas/tests/plotting/test_frame.py#L3437 it hits the `elif is_datetime64tz_dtype` path. This was suggested in the original PR so I also added this as test case as the reviewer there obviously thought it was supposed to be supported.

Reply:
See also my other comment for the test data.